File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/printStars.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/printStars
 *
 * @author Alex Golub
 * @since 09-May-21, 2:43 PM
 */
function printStars(n) {
  if (n < 0) {
    throw 'argument must be non-negative';
  }
  console.log(helper(n, ''));
}

function helper(n, s) {
  if (n === 0) {
    return;
  } if (n === 1) {
    return s + '*';
  } else {
    return helper(n - 1, s + '*')
  }
}

printStars(5);

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/printStars.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/printStars
 *
 * @author Alex Golub
 * @since 09-May-21, 2:41 PM
 */
class printStars {
  public static void printStars(int n) {
    if (n == 0) {
      return;
    } else if (n == 1) {
      System.out.print("*");
    } else {
      System.out.print("*");
      printStars(n - 1);
    }
  }
}
